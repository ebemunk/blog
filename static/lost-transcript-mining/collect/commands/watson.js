/*
tone analyzer
	first 1000 calls free
	up to 128kb text or 1000 sentences

personality insights
	first 100 calls free
	20mb

alchemy api
	free plan, 1000 api plans per day

*/
