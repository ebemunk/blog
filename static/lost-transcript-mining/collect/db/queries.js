export function totalLinesByChar() {
	return `
		select char_name, count(*) as lines
		from dialog
		where type='dialog' and char_name is not null
		group by char_name
		order by lines desc
	;`
}

export function allLines(char_name) {
	return `
		select string_agg(line, E'\n' order by season, episode, seq) as text
		from dialog
		where type='dialog' and char_name='${char_name}'
		group by char_name
	;`
}

export function episodeText(season, episode) {
	return `
		select string_agg(char_name || ': ' || line, E'\n' order by seq) as text
		from dialog
		where type='dialog' and season=${season} and episode=${episode}
		group by season,episode
		order by season,episode
	;`
}
