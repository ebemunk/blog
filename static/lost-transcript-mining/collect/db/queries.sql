-- total lines by char
select char_name, count(*) as lines
from dialog
where type='dialog' and char_name is not null
group by char_name
order by lines desc
;

-- total lines per char per episode per season
select season, episode, char_name, count(*) as lines
from dialog
where type='dialog'
group by season, episode, char_name
order by season, episode, lines desc
;

-- char per episode per season
select distinct season, episode, char_name from dialog
where type='dialog'
order by season, episode, char_name asc
;

-- all spoken dialog prefixed with char name by episode
select season, episode, string_agg(char_name || ': ' || line, E'\n' order by seq)
from dialog
where type='dialog'
group by season,episode
order by season,episode
;

-- all dialog for each char ordered chronologically
select char_name, string_agg(line, E'\n' order by season, episode, seq)
from dialog
where type='dialog'
group by char_name
;

-- chars in every scene
select season, episode, act, scene, array_agg(distinct char_name)
from dialog
where type='dialog'
group by season, episode, act, scene
;