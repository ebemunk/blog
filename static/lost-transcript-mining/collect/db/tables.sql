-- dialog table
CREATE TABLE public.dialog
(
  id SERIAL NOT NULL,
  season integer NOT NULL,
  episode integer NOT NULL,
  seq integer NOT NULL,
  act integer NOT NULL,
  scene integer NOT NULL,
  flashback boolean NOT NULL,
  flashsideways boolean NOT NULL,
  type character varying NOT NULL,
  char_name character varying DEFAULT NULL,
  line text DEFAULT NULL,
  directions json DEFAULT '[]'::json,
  raw text DEFAULT NULL
)

-- watson personality insights
CREATE TABLE public.personality
(
  id SERIAL NOT NULL,
  char_name character varying DEFAULT NULL,
  profile json DEFAULT '{}'::json
)

-- watson tone analyzer for all episodes
CREATE TABLE public.tone
(
  season integer NOT NULL,
  episode integer NOT NULL,
  document_tone json DEFAULT '{}'::json,
  sentences_tone json DEFAULT '[]'::json
)
