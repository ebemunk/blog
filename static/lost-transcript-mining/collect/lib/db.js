export function insertObj(obj, table = 'dialog') {
	const cols = Object.keys(obj)
	.join(',')
	const valOrder = Object.keys(obj)
	.map((v, i) => `$${i+1}`)
	.join(',')
	const sql = `
		INSERT INTO ${table}
		(${cols})
		VALUES
		(${valOrder})
	`
	return [
		sql,
		Object.values(obj)
	]
}
