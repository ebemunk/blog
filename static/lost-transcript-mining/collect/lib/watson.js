import { config } from 'dotenv'
import {
	tone_analyzer,
	personality_insights,
} from 'watson-developer-cloud'
import R from 'ramda'
import Promise from 'bluebird'

import {
	queries
} from '../db'

config()

const {
	TONE_ANALYZER_USERNAME,
	TONE_ANALYZER_PASSWORD,
	PERSONALITY_INSIGHTS_USERNAME,
	PERSONALITY_INSIGHTS_PASSWORD
} = process.env

const toneAnalyzer = Promise.promisifyAll(tone_analyzer({
	version: 'v3',
	version_date: '2016-05-19',
	username: TONE_ANALYZER_USERNAME,
	password: TONE_ANALYZER_PASSWORD
}))

export const personalityInsights = Promise.promisifyAll(personality_insights({
	version: 'v3',
	version_date: '2016-10-19',
	username: PERSONALITY_INSIGHTS_USERNAME,
	password: PERSONALITY_INSIGHTS_PASSWORD
}))

export async function charProfile(pool, char_name) {
	const result = await pool.query(queries.allLines(char_name))
	const profile = await personalityInsights.profileAsync({
		text: result.rows[0].text,
		consumption_preferences: true
	})
	return profile
}

export async function episodeTone(pool, season, episode) {
	return R.pipeP(
		sql => pool.query(sql),
		R.prop('rows[0].text'),
	)(queries.episodeText(season, episode))
}
