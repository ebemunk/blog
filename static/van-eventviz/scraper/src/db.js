import { Pool } from 'pg'
import { config } from 'dotenv'
import Promise from 'bluebird'

config()

const { DB_USER, DB_PASSWORD, DB_HOST, DB_PORT, DB_NAME } = process.env

const pool = new Pool({
  user: DB_USER,
  password: DB_PASSWORD,
  host: DB_HOST,
  port: DB_PORT,
  database: DB_NAME,
  Promise,
})

export function getPool() {
  return pool
}

export function insertObj(obj, table) {
  const cols = Object.keys(obj).join(',')
  const valOrder = Object.keys(obj)
    .map((v, i) => `$${i + 1}`)
    .join(',')
  const sql = `
		INSERT INTO ${table}
		(${cols})
		VALUES
		(${valOrder})
	`
  return [sql, Object.values(obj)]
}
