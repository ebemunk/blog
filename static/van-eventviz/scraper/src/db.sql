drop table events;
create table events (
  id character varying primary key,
  evt json not null
  details json,
  geo json
);
