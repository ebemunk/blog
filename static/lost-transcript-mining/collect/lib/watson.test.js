import sinon from 'sinon'
import * as watson from './watson'
import * as db from '../db'

describe('watson', () => {
	const sandbox = sinon.sandbox.create()

	afterEach(() => sandbox.restore())

	describe('charProfile', () => {
		it('should make it happen', async () => {
			const queryStub = jest.fn(() => Promise.resolve({
				rows: [
					{ text: 'blabla' }
				]
			}))
			const profileStub = jest.fn(() => Promise.resolve({}))
			sandbox
			.stub(watson.personalityInsights, 'profileAsync')
			.callsFake(profileStub)

			await watson.charProfile({ query: queryStub }, 'JACK')

			expect(queryStub).toHaveBeenCalledTimes(1)
			expect(queryStub.mock.calls).toMatchSnapshot()
			expect(profileStub).toHaveBeenCalledTimes(1)
			expect(profileStub.mock.calls).toMatchSnapshot()
		})
	})

	describe('episodeTone', () => {
		it('should make it happen', async () => {
			const kek = await watson.episodeTone(db.getPool(), 1, 1)
			console.log(kek);
		})
	})
})
