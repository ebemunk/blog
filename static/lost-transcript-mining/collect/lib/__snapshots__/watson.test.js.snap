// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`watson charProfile should make it happen 1`] = `
Array [
  Array [
    "
		select string_agg(line, E'
' order by season, episode, seq) as text
		from dialog
		where type='dialog' and char_name='JACK'
		group by char_name
	;",
  ],
]
`;

exports[`watson charProfile should make it happen 2`] = `
Array [
  Array [
    Object {
      "consumption_preferences": true,
      "text": "blabla",
    },
  ],
]
`;
